### ISSUE TITLE
Upgrade to Serde 1.0

### ISSUE THREAD

**Issue by kardeiz** [samscott89/serde_qs#3]
Hi, 

I made a naive attempt to update `serde_qs` to Serde 1.0 (released last week). I got everything to compile, but got stuck at making tests pass:

```
thread 'deserialize_struct' panicked at 'called `Result::unwrap()` on an `Err` value: 
Error { err: "invalid type: string \"42\", expected u8" }'
```

I didn't want to make a failing PR, but you can check out the changes here: https://github.com/samscott89/serde_qs/compare/master...kardeiz:serde-update. 

I mostly just followed the compiler's instructions in updating to the new API&mdash;

**Comment by kardeiz** [comment]
If you like, I can go ahead and make a PR with what I have, or if you see what might be wrong right away, I'd be happy to edit and make a PR.

**Comment by samscott89** [comment]
Hey!
Thanks for having a go at this. I was going to sit down and do this myself tomorrow so that's a huge help.

I'll look into this more tomorrow, but the problem seems to be that before serde was happy to convert a string into a Primtiive, whereas now it is not.

**Comment by samscott89** [comment]
Right, so the issues were due to this PR: https://github.com/serde-rs/serde/pull/839. Which makes it fairly clear why the previous method could be problematic.

On the other hand, the example given makes no sense for `serde_qs`:
```rust
    #[derive(Deserialize, Debug, PartialEq)]
    #[serde(untagged)]
    enum E {
        B(bool),
        S(String),
    }

    #[derive(Deserialize, Debug, PartialEq)]
    struct Query {
        e: E,    
    }

    let params = "e=true";
    let rec_params: Query = qs::from_str(params).unwrap();
    assert_eq!(rec_params, Query { e: E::B(true) }); // this fails
```
since there is nothing distinguishing strings from primitives (i.e. `"42" vs 42`, both would be written as `value=42`).

I would guess this makes it not suitable for untagged enums (it's not a self-describing format, and will always deserialize to a string).

I'll try to write something to that effect in the docs somewhere. I'll push something including your commit shortly. Thanks again for starting the work on this!

**Comment by samscott89** [comment]
Hey @kardeiz, what do you think of the additions?

**Comment by kardeiz** [comment]
Looks good and thanks for the background and details!

Two issues:

Parsing errors should return an error, not panic.

This doesn't currently support maps with primitive-type keys. 

I'm about to make a PR to fix these. It adds two very short tests to test these two issues.

**Comment by samscott89** [comment]
Okay, merged in 0.1.0 and published. Thanks for the work!
